Validation/RPCRecHits/src/RPCRecHitValid.cc

```diff
@@ -31,6 +29,9 @@ RPCRecHitValid::RPCRecHitValid(const edm::ParameterSet &pset) {
   muonToken_ = consumes<reco::MuonCollection>(pset.getParameter<edm::InputTag>("muon"));
 
   subDir_ = pset.getParameter<std::string>("subDir");
+
+  rpcGeomToken_ = esConsumes();
+  rpcGeomTokenInRun_ = esConsumes<edm::Transition::BeginRun>();
```

Comment:
I am not sure why you need here a Transition since bookHistograms are not a real beginRun, though MEs are booked at the beginning, I guess it does not harm

Reply:
@makortel since bookHistograms is not a genuine beginRun function, could this type of lines (or its omision) introduce differences in MEs when the code is run in the multi-thread case? Thanks

Reply:
`bookHistograms()` is called from `beginRun()` so framework sees it as code run at `beginRun` and therefore `BeginRun` transition is correct. Framework throws an exception if the token is used in a different transition than the consumes was declared for.

Reply:
And the other way around? I mean, if BeginRun transition is set (but not needed) instead of a simple esConsumes() is set, would it harm? Harm = likely to produce wrong results?

Reply:
I'm not sure I understand what you mean with "if BeginRun transition is set but not needed". The default transition for `esConsumes` is Event transition (i.e. in `analyze()`/`produce()`/`filter()`). Using a token declared for BeginRun transition at Event transitions (or vice versa) leads to an exception.

Reply:
Thanks, understood!